lib/jekyll/static_file.rb

```diff
@@ -100,5 +101,40 @@ def to_liquid
         "path"          => File.join("", relative_path)
       }
     end
+
+    # Applies a similar URL-building technique as Jekyll::Document that takes
+    # the collection's URL template into account. The default URL template can
+    # be overriden in the collection's configuration in _config.yml.
+    def url
+      return @url unless @url.nil?
+      if @collection.nil?
+        @url = self.relative_path
+      else
+        @url = ::Jekyll::URL.new({
+          template:  @collection.url_template,
+          placeholders: {
```

Comment:
would you please create a separate method which creates the `placeholders` hash?